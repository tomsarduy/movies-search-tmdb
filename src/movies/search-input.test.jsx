import React from "react";
import SearchInput from "./search-input.jsx";

describe("OrderByInput", () => {
  it("renders correctly", () => {
    const component = shallow(<SearchInput onChange={() => null} />);
    expect(component).toMatchSnapshot();
  });

  it("should call onChange with the selected value on change", () => {
    const onChangeMock = jest.fn();
    const target = { value: "typing..." };
    const component = shallow(<SearchInput onChange={onChangeMock} />);
    component.find("input").simulate("change", { target });
    expect(onChangeMock).toHaveBeenCalledTimes(1);
    expect(onChangeMock).toHaveBeenCalledWith({ target });
  });
});
