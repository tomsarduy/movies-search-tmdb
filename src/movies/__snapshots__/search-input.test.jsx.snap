// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OrderByInput renders correctly 1`] = `
<input
  id="movie-search"
  name="q"
  onChange={[Function]}
  placeholder="Search movies..."
  type="search"
/>
`;
